src/m365/pp/base/PowerPlatformCommand.spec.ts

```diff
@@ -0,0 +1,85 @@
+import * as assert from 'assert';
+import * as sinon from 'sinon';
+import appInsights from '../../../appInsights';
+import auth from '../../../Auth';
+import { Logger } from '../../../cli';
+import Utils from '../../../Utils';
+import PowerPlatformCommand from './PowerPlatformCommand';
+
+class MockCommand extends PowerPlatformCommand {
+  public get name(): string {
+    return 'mock';
+  }
+
+  public get description(): string {
+    return 'Mock command';
+  }
+
+  public commandAction(logger: Logger, args: any, cb: () => void): void {
+    cb();
+  }
+
+  public commandHelp(): void {
+  }
+}
+
+describe('PowerPlatformCommand', () => {
+  before(() => {
+    sinon.stub(appInsights, 'trackEvent').callsFake(() => { });
+  });
+
+  afterEach(() => {
+    Utils.restore(auth.restoreAuth);
+  });
+
+  after(() => {
+    Utils.restore(appInsights.trackEvent);
+  });
+
+  it('doesn\'t execute command when not logged in', (done) => {
```

Comment:
Since we're not changing the auth behavior of the base command, we only need the test for resource